javav2/example_code/kendra/src/main/java/com/example/kendra/CreateIndexAndDataSourceExample.java

```diff
@@ -0,0 +1,197 @@
+//snippet-sourcedescription:[CreateIndexAndDataSourceExample.java demonstrates how to create an Amazon Kendra index and data source.]
+//snippet-keyword:[SDK for Java v2]
+//snippet-keyword:[Code Sample]
+//snippet-service:[Amazon Kendra]
+//snippet-sourcetype:[full-example]
+//snippet-sourcedate:[03/10/2022]
+//snippet-sourceauthor:[scmacdon - aws]
+
+/*
+   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0
+*/
+
+package com.example.kendra;
+
+// snippet-start:[kendra.java2.index.import]
+import java.util.concurrent.TimeUnit;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.kendra.KendraClient;
+import software.amazon.awssdk.services.kendra.model.CreateIndexResponse;
+import software.amazon.awssdk.services.kendra.model.DescribeIndexRequest;
+import software.amazon.awssdk.services.kendra.model.DescribeIndexResponse;
+import software.amazon.awssdk.services.kendra.model.KendraException;
+import software.amazon.awssdk.services.kendra.model.IndexStatus;
+import software.amazon.awssdk.services.kendra.model.CreateDataSourceRequest;
+import software.amazon.awssdk.services.kendra.model.CreateIndexRequest;
+import software.amazon.awssdk.services.kendra.model.DataSourceType;
+import software.amazon.awssdk.services.kendra.model.DataSourceConfiguration;
+import software.amazon.awssdk.services.kendra.model.S3DataSourceConfiguration;
+import software.amazon.awssdk.services.kendra.model.CreateDataSourceResponse;
+import software.amazon.awssdk.services.kendra.model.DescribeDataSourceRequest;
+import software.amazon.awssdk.services.kendra.model.DataSourceStatus;
+import software.amazon.awssdk.services.kendra.model.DescribeDataSourceResponse;
+import software.amazon.awssdk.services.kendra.model.StartDataSourceSyncJobResponse;
+import software.amazon.awssdk.services.kendra.model.StartDataSourceSyncJobRequest;
+// snippet-end:[kendra.java2.index.import]
+
+/**
+ * To run this Java V2 code example, ensure that you have set up your development environment, including your credentials.
+ *
+ * For information, see this documentation topic:
+ *
+ * https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
+ *
+ */
+public class CreateIndexAndDataSourceExample {
+
+    public static void main(String[] args) {
+
+         final String usage = "\n" +
+                    "Usage:\n" +
+                    "    <indexDescription> <indexName> <indexRoleArn> <dataSourceRoleArn> <dataSourceName> <dataSourceDescription> <s3BucketName>\n\n" +
+                    "Where:\n" +
+                    "    indexDescription - a description for the index.\n" +
+                    "    indexName - the name for the new index.\n" +
+                    "    indexRoleArn - an Identity and Access Management (IAM) role that gives Amazon Kendra permissions to access your Amazon CloudWatch logs and metrics.\n\n" +
```

Comment:
"an Identity and Access Management (IAM) role" --> "An AWS Identity and Access Management (IAM) role"